suite-common/connect/src/actions/wrapMethods.ts

```diff
@@ -0,0 +1,41 @@
+import TrezorConnect from '@trezor/connect';
+import { AnyAction, createAsyncThunk, Dispatch } from '@reduxjs/toolkit';
+
+export const wrapMethods = createAsyncThunk<
+    void,
+    {
+        lockDevice: (payload: boolean) => AnyAction;
+    },
+    {
+        dispatch: Dispatch;
+    }
+>('wrapMethods', ({ lockDevice }, { dispatch }) => {
+    const wrappedMethods = [
+        'getFeatures',
+        'getDeviceState',
+        'getAddress',
+        'ethereumGetAddress',
+        'rippleGetAddress',
+        'cardanoGetAddress',
+        'applySettings',
+        'changePin',
+        'pushTransaction',
+        'ethereumSignTransaction',
+        'signTransaction',
+        'rippleSignTransaction',
+        'cardanoSignTransaction',
+        'backupDevice',
+        'recoveryDevice',
+    ] as const;
+    wrappedMethods.forEach(key => {
+        // typescript complains about params and return type, need to be "any"
+        const original: any = TrezorConnect[key];
+        if (!original) return;
+        (TrezorConnect[key] as any) = async (params: any) => {
+            dispatch(lockDevice(true));
```

Comment:
Maybe we should make all these patches redux agnostic. What do you think?

 if we make this just function then we can do something like 

```tsx
const lockDeviceConnectPatch = ({onDeviceLock, onDeviceUnlock}) => {
...
onDeviceLock();
const result = await original(params);
onDeviceUnlock();
...

// usage in init action
lockDeviceConnectPatch({
  onDeviceLock: () => dispatch({lockDevice(true)}),
  onDeviceUnlock: () => dispatch({lockDevice(false)}),
})
```

Reply:
I agree. It is definitely a more elegant solution! https://github.com/trezor/trezor-suite/pull/5760/commits/aca99f3d8491e5725750f30a77cee03823be443d#diff-f8b4494e69ec35a7431e26e9f0928a1da250bb00abaaebf905e91fa19de47bdc